pkg/controllers/raycluster_controller.go

```diff
@@ -337,6 +361,56 @@ func desiredOAuthSecret(cluster *rayv1.RayCluster, r *RayClusterReconciler) *cor
 	// Create a Kubernetes secret to store the cookie secret
 }
 
+func desiredNetworkPolicy(cluster *rayv1.RayCluster, kubeRayNamespaces []string) *networkingapply.NetworkPolicyApplyConfiguration {
+	return networkingapply.NetworkPolicy(cluster.Name, cluster.Namespace).
+		WithLabels(map[string]string{"ray.io/cluster-name": cluster.Name}).
+		WithSpec(networkingapply.NetworkPolicySpec().
+			WithPodSelector(metav1apply.LabelSelector().WithMatchLabels(map[string]string{"ray.io/cluster": cluster.Name, "ray.io/node-type": "head"})).
+			WithIngress(
+				networkingapply.NetworkPolicyIngressRule().
+					WithPorts(
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(6379)),
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(10001)),
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(8080)),
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(8265)),
+					).WithFrom(
+					networkingapply.NetworkPolicyPeer().WithPodSelector(metav1apply.LabelSelector()),
+				),
+				networkingapply.NetworkPolicyIngressRule().
+					WithFrom(
+						networkingapply.NetworkPolicyPeer().WithPodSelector(metav1apply.LabelSelector().
+							WithMatchLabels(map[string]string{"app.kubernetes.io/component": "kuberay-operator"})).
+							WithNamespaceSelector(metav1apply.LabelSelector().
+								WithMatchExpressions(metav1apply.LabelSelectorRequirement().
+									WithKey(corev1.LabelMetadataName).
+									WithOperator(metav1.LabelSelectorOpIn).
+									WithValues(kubeRayNamespaces...)))).
+					WithPorts(
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(8265)),
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(10001)),
+					),
+				networkingapply.NetworkPolicyIngressRule().
+					WithPorts(
+						networkingapply.NetworkPolicyPort().WithProtocol(corev1.ProtocolTCP).WithPort(intstr.FromInt(8080)),
+					).
+					WithFrom(
+						networkingapply.NetworkPolicyPeer().WithNamespaceSelector(metav1apply.LabelSelector().
+							WithMatchExpressions(metav1apply.LabelSelectorRequirement().
+								WithKey(corev1.LabelMetadataName).
+								WithOperator(metav1.LabelSelectorOpIn).
+								WithValues("openshift-user-workload-monitoring"))),
```

Comment:
I think that should be `openshift-monitoring` instead.

Reply:
does SRE monitoring need access redhat-ods-applications (your ray controller) ? or this is for UWM?